ext/standard/math.c

```diff
@@ -126,6 +126,24 @@ static inline double php_round_helper(double value, int mode) {
 
 			return integral;
 
+		case PHP_ROUND_CEILING:
+			if (value > 0.0) {
+				return fractional == 0 ? integral : integral + 1.0;
+			}
+			return integral;
```

Comment:
```suggestion
			if (value > 0.0 && fractional > 0.0) {
				return integral + 1.0;
			}

			return integral;
```
or

```suggestion
			if (value > 0.0) {
				if (fractional > 0.0) {
					return integral + 1.0;
				}
			}

			return integral;
```
or

```suggestion
			if (fractional > 0.0) {
				if (value > 0.0) {
					return integral + 1.0;
				}
			}

			return integral;
```

I feel each of these would be clearer. I'll probably take a look at the assembly later, time permitting.

Reply:
Thank you. Done

Reply:
Checked the assembly now: The first two result in the same assembly. The third is slightly different (which somewhat makes sense, because of `&&`'s shortcutting semantics). I don't think either one is strictly better than the other, so this is fine.